Mathlib/Topology/ContinuousFunction/FunctionalCalculus.lean

```diff
@@ -0,0 +1,676 @@
+`cfc a (g ∘ f) = cfc (cfc a f) g` under suitable hypotheses on `a`, `f` and `g`. Note that this
+theorem is nearly impossible to state nicely in terms of `cfcSpec` (see `cfcSpec_comp`). An
+additional advantage of the unbundled approach is that expressions like `fun x : R ↦ x⁻¹` are valid
+arguments to `cfc a`, and a bundled continuous counterpart can only make sense when the spectrum of
+`a` does not contain zero and when we have an `⁻¹` operation on the domain.
+
+A reader familiar with C⋆-algebra theory may be somewhat surprised at the level of abstraction here.
+For instance, why not require `A` to be an actual C⋆-algebra? Why define separate continuous
+functional calculi for `R := ℂ`, `ℝ` or `ℝ≥0` instead of simply using the continuous functional
+calculus for normal elements? The reason for both can be explained with a simple example,
+`A := Matrix n n ℝ`. In Mathlib, matrices are not equipped with a norm (nor even a metric), and so
+requiring `A` to be a C⋆-algebra is far too stringent. Likewise, `A` is not a `ℂ`-algebra, and so
+it is impossible to consider the `ℂ`-spectrum of `a : Matrix n n ℝ`.
+
+There is another, more practical reason to define separate continuous functional calculi for
+different scalar rings. It gives us the ability to use functions defined on these types, and the
+algebra of functions on them. For example, in `ℝ` it is quite natural to consider the functions
+`(·⁺ : ℝ → ℝ)` and `(·⁻ : ℝ → ℝ)` because the collection of such functions forms a lattice ordered
+group. If `a : A` is selfadjoint, and we define `a⁺ := cfc a (·⁺ : ℝ → ℝ)`, and likewise for `a⁻`,
+then the properties `a⁺ * a⁻ = 0 = a⁻ * a⁺` and `a = a⁺ - a⁻` are trivial consequences of the
+corresponding facts for functions. In contrast, if we had to do this for functions on `ℂ`, the
+proofs of these facts would be much more cumbersome.
+
+## Example
+
+The canonical example of the continuous functional calculus is when `A := Matrix n n ℂ`, `R := ℂ`
+and `p := IsStarNormal`. In this case, `spectrum ℂ a` consists of the eigenvalues of the normal
+matrix `a : Matrix n n ℂ`, and, because this set is discrete, any function is continuous on the
+spectrum. The continuous functional calculus allows us to make sense of expressions like `log a`
+(`:= cfc a log`), and when `0 ∉ spectrum ℂ a`, we get the nice property `exp (log a) = a`, which
+arises from the composition property `cfc (cfc log a) exp = cfc (exp ∘ log) a = cfc id a = a`, since
+`exp ∘ log = id` *on the spectrum of `a`*. Of course, there are other ways to make sense of `exp`
+and `log` for matrices (power series), and these agree with the continuous functional calculus.
+In fact, given `f : C(spectrum ℂ a, ℂ)`, `cfc a f` amounts to diagonalizing `a` (possible since `a`
+is normal), and applying `f` to the resulting diagonal entries. That is, if `a = u * d * star u`
+with `u` a unitary matrix and `d` diagonal, then `cfc a f = u * (d.map f) * star u`.
```

Comment:
Just noting the brackets are not necessary.
```suggestion
with `u` a unitary matrix and `d` diagonal, then `cfc a f = u * d.map f * star u`.
```
Are you putting them for clarity? I'm not sure it helps much.